client/src/main/java/io/pravega/client/segment/impl/SegmentInputStreamImpl.java

```diff
@@ -270,6 +270,12 @@ public int bytesInBuffer() {
         return result;
     }
 
+    @Override
+    @Synchronized
+    public boolean isEndOfSegment() {
```

Comment:
I think you want to also check to make sure there is nothing in the buffer

Reply:
Done.